version: "3.8"

services:
    postgresql:
        image: bitnami/postgresql:16.4.0
        container_name: postgresql
        #user: ${MY_USER_ID}:${MY_GROUP_ID}
        volumes:
            - postgresql_volume:/var/lib/postgresql
            - ./docker/PostgreSQL/init:/docker-entrypoint-initdb.d
        networks:
            - matcha-network
        environment:
            POSTGRESQL_ROOT_PASSWORD: ${POSTGRESQL_ROOT_PASSWORD}
            POSTGRESQL_USER: ${POSTGRESQL_USER}
            POSTGRESQL_PASSWORD: ${POSTGRESQL_PASSWORD}
            POSTGRESQL_DATABASE: ${POSTGRESQL_DATABASE}
        restart: on-failure
        expose:
            - "5432"

    backend:
        container_name: backend
        build: 
            context: ./docker/Backend
            dockerfile: Dockerfile
        volumes:
            - ./Backend:/Backend
        env_file: .env
        ports:
          - "3001"

    frontend:
        container_name: frontend
        build: 
            context: ./docker/Frontend
            dockerfile: Dockerfile
        volumes:
            - ./Frontend:/Frontend
        env_file: .env
        ports:
          - "3000:80"

volumes:
    postgresql_volume:
        driver: local
        driver_opts:
            type: 'none'
            o: 'bind'
            device: '${POSTGRESQL_VOLUME_PATH}'

networks:
    matcha-network:
        driver: bridge